socketio/src/socket.rs

```diff
@@ -70,54 +87,75 @@ impl Socket {
         Ok(())
     }
 
-    /// Emits to certain event with given data. The data needs to be JSON,
-    /// otherwise this returns an `InvalidJson` error.
+    /// Emits to certain event with given data.
     pub fn emit(&self, nsp: &str, event: Event, data: Payload) -> Result<()> {
-        let socket_packet = self.build_packet_for_payload(data, event, nsp, None)?;
-
+        let socket_packet = Self::build_packet_for_payload(data, Some(event), nsp, None, false)?;
         self.send(socket_packet)
     }
 
     /// Returns a packet for a payload, could be used for bot binary and non binary
     /// events and acks. Convenance method.
     #[inline]
-    pub(crate) fn build_packet_for_payload<'a>(
-        &'a self,
+    pub(crate) fn build_packet_for_payload(
         payload: Payload,
-        event: Event,
-        nsp: &'a str,
+        event: Option<Event>,
```

Comment:
nit: when is event none? Only usage I can see is in the tests

Reply:
I found  `client.emit(Buffer.from([4, 5, 6]));` in socket-io.js, I thought event is optional that time. But I lookup the socketio dos, event is not optional. Does this mean the event can be a valid payload?

Reply:
right now in `rust-socketio` if only receive 1 item in packet.data, it will be treat as payload, not event.

https://github.com/1c3t3a/rust-socketio/blob/4ac27c7158a89375c503164857d86b1a1167116c/socketio/src/client/raw_client.rs#L325-L354

Reply:
Where did you find that? Specification only mentions, that when no event is specified, "Message" should be used. So having an `Option<Event>` would not make sense in that case.

Reply:
@1c3t3a I found it here https://github.com/1c3t3a/rust-socketio/blob/4ac27c7158a89375c503164857d86b1a1167116c/ci/socket-io.js#L41